### DISCUSSION TITLE
PRINT_END modification for cooling

### DISCUSSION THREAD

**Discussion by tkunchick** [jschuh/klipper-macros#218] [Ideas]
I seen this on the K1 MAX from Creailty and was wondering if you could modify the print_end to do something similar.

Basically when the heaters shuts off it turns the fans up high until the extruder is below 40 degrees then shuts the fans down.

There is a call out in the print_start to end the timer (WAIT_TEMP_END) incase you start a new print before cool down has completed.

Maybe add something to the _km_options to enable / disable functionality.

```
[delayed_gcode wait_temp]
# initial_duration:2.
gcode:
  {% set cur_temp = printer.extruder.temperature %}
  {% if cur_temp|int < 40 %}
    {action_respond_info("end_temp!\n")}
    WAIT_TEMP_END
  {% else %}
    {action_respond_info("cur_temp = %s \n" % (cur_temp))}
    UPDATE_DELAYED_GCODE ID=wait_temp DURATION=5
  {% endif %}

[gcode_macro WAIT_TEMP_START]
gcode:
  UPDATE_DELAYED_GCODE ID=wait_temp DURATION=1
  M106 P0 S255
  M106 P2 S255

[gcode_macro WAIT_TEMP_END]
gcode:
  UPDATE_DELAYED_GCODE ID=wait_temp DURATION=0
  M106 P0 S0
  M106 P2 S0
```

Would like to say I am glad I stumbled across your macros. Saved me a huge headache trying to learn and setup Klipper on my rasp pi for my ENDER3 V2 NEO

**Comment by geirskogul** [comment]
I really like this, even years later, and put it on all my machines for turnaround
